esp-hal-rom/build.rs

```diff
@@ -0,0 +1,79 @@
+use std::{error::Error, path::Path};
+
+use esp_metadata::{Chip, Config};
+
+#[macro_export]
+macro_rules! assert_unique_features {
+    ($($feature:literal),+ $(,)?) => {
+        assert!(
+            (0 $(+ cfg!(feature = $feature) as usize)+ ) <= 1,
+            "Exactly zero or one of the following features must be enabled: {}",
+            [$($feature),+].join(", ")
+        );
+    };
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // Ensure that exactly one chip has been specified:
+    let chip = Chip::from_cargo_feature()?;
+
+    // Log and defmt are mutually exclusive features. The main technical reason is
+    // that allowing both would make the exact panicking behaviour a fragile
+    // implementation detail.
+    assert_unique_features!("log-04", "defmt");
```

Comment:
FWIW: One could do like embassy usually does by placing this in lib.rs or the like. Removes the need for the `assert_unique_features` macro too.
```rust
#[cfg(all(feature = "defmt", feature = "log"))]
compile_error!("You may not enable both `defmt` and `log` features.");
```